x-pack/plugins/apm/server/routes/settings/custom_link/create_custom_link_index.ts

```diff
@@ -12,19 +12,29 @@ import {
   Mappings,
 } from '@kbn/observability-plugin/server';
 import { APM_CUSTOM_LINK_INDEX } from '../apm_indices/get_apm_indices';
+import { ApmFeatureFlags } from '../../../../common/apm_feature_flags';
 
 export const createApmCustomLinkIndex = async ({
   client,
   logger,
+  featureFlags,
 }: {
   client: ElasticsearchClient;
   logger: Logger;
+  featureFlags: ApmFeatureFlags;
 }) => {
   return createOrUpdateIndex({
     index: APM_CUSTOM_LINK_INDEX,
     client,
     logger,
     mappings,
+    settings: featureFlags.fastRefreshAvailable
+      ? {
+          index: {
+            fast_refresh: true,
+          },
+        }
+      : {},
```

Comment:
This will fail if it tried to add the `fast_refresh` setting to an existing index that doesn't already have it. Since this only applies to serverless it will probably not be a problem because the APM system indicies will be "born" with the setting, and therefore won't need to be updated. But it could bite us in the future if we want to enable the setting for non-serverless.

Reply:
We are just doing this for the index creation, not adding the settings value for updates, check x-pack/plugins/observability/server/utils/create_or_update_index.ts

Reply:
I'm not seeing anything in that file that indicates you are restricting this to new indices only. Maybe I'm overlooking it. Can you point me to it exactly?

The code I'm referring to is this:

https://github.com/elastic/kibana/blob/b972b07182ede880bec35f0b52b6a3f7dbe3f537/x-pack/plugins/observability/server/utils/create_or_update_index.ts#L41-L53

If the index exists it is updated, else created.

Reply:
We are just adding the settings in case of create, so fast_refresh won't be added if the index already exists for apm.custom.links